packages/containers/src/SidePanel/SidePanel.connect.js

```diff
@@ -1,39 +1,93 @@
 import get from 'lodash/get';
 import cmf, { cmfConnect } from '@talend/react-cmf';
-import { withRouter } from 'react-router';
+import zip from 'lodash/zip';
 import Container, { DEFAULT_STATE } from './SidePanel.container';
+import { ACTION_TYPE_LINK } from './constants';
+
+const cache = {};
+
+function splitPath(path) {
+	return path.replace(/^(\/)/, '').split('/');
+}
+
+function isBasePathOf(pathToCheck, fullPath) {
+	const pathToCheckSplit = splitPath(pathToCheck);
+	const fullPathSplit = splitPath(fullPath);
+
+	return zip(pathToCheckSplit, fullPathSplit)
+		.filter(([pathToCheckPart]) => pathToCheckPart !== undefined)
+		.every(([pathToCheckPart, fullPathPart]) => pathToCheckPart === fullPathPart);
+}
+
+const getActionsWrapped = actions =>
+	actions.map(action => {
+		if (!action.href) {
+			return action; // do not change ref
+		}
+		if (action.href && !action.onClick && !action.onClickDispatch && !action.onClickActionCreator) {
+			return {
+				...action,
+				onClick: event => {
+					event.preventDefault();
+					event.stopPropagation();
+				},
+				onClickDispatch: {
+					type: ACTION_TYPE_LINK,
+					cmf: {
+						routerPush: action.href,
+					},
+				},
+			};
+		}
+		return action;
+	});
+
+const getSelectedAction = (currentRoute, actions) =>
+	actions.find(action => action.href && isBasePathOf(action.href, currentRoute));
 
 export function mapStateToProps(state, ownProps) {
 	const props = {};
+	const currentRoute = cmf.selectors.router.getPath(state);
 	if (ownProps.actionIds) {
-		props.actions = ownProps.actionIds.map(id => {
-			const action = { actionId: id };
-			const info = cmf.action.getActionInfo(
-				{
-					registry: cmf.registry.getRegistry(),
-					store: {
-						getState: () => state,
+		const cacheKey = `${currentRoute}-${ownProps.actionIds.join('#')}`;
+		if (!cache[cacheKey]) {
+			cache[cacheKey] = ownProps.actionIds.map(id => {
+				const action = { actionId: id };
+				const info = cmf.action.getActionInfo(
+					{
+						registry: cmf.registry.getRegistry(),
+						store: {
+							getState: () => state,
+						},
 					},
-				},
-				id,
-			);
-			action.label = info.label;
-			action.id = info.id;
-			let route = get(info, 'payload.cmf.routerReplace');
-			if (!route) {
-				route = get(info, 'payload.cmf.routerPush');
-			}
-			if (!route) {
-				route = get(info, 'href');
-			}
-			if (route) {
-				const currentRoute = ownProps.location.pathname;
-				if (currentRoute.indexOf(route) !== -1) {
-					action.active = true;
+					id,
+				);
+				action.label = info.label;
+				action.id = info.id;
+				let route = get(info, 'payload.cmf.routerReplace');
+				if (!route) {
+					route = get(info, 'payload.cmf.routerPush');
 				}
-			}
-			return action;
-		});
+				if (!route) {
+					route = get(info, 'href');
+				}
+				if (route) {
+					if (currentRoute.indexOf(route) !== -1) {
+						action.active = true;
+					}
```

Comment:
You could use `includes()` :

`if (currentRoute.includes(route)` or `action.active = currentRoute.includes(route);`

Reply:
I have finally moved to the new check which work better